apicast/src/proxy.lua

```diff
@@ -361,6 +362,15 @@ function _M:access(service)
 
   local credentials, err = service:extract_credentials()
 
+  if keycloak then
```

Comment:
I think it was there in a previous iteration and we moved it because `extract_credentials` is just reading the credentials from the request, not parsing/validating them.

I think you suggested moving to `apicast.lua` at the time, but I couldn't see where, so I put in `proxy.lua` instead